### ISSUE TITLE
Colorbars (intermittently) not including numerical values

### ISSUE THREAD

**Issue by claytonstrawn** [yt-project/yt#1639]
![no-ticks_projection_z_h_p0_number_density](https://user-images.githubusercontent.com/33767568/33495499-870c1b20-d67b-11e7-876d-dc094c31cd7c.png)
![ticks](https://user-images.githubusercontent.com/33767568/33495500-8724b5a4-d67b-11e7-9037-c5f3321e4558.png)

Sometimes, when I am plotting graphs with a colorbar, no numerical values are added to the colorbar, making it unusable. Probably it has something to do with the dynamic range, but I haven't always seen it be consistently better on smaller or larger datasets.

The file I used was uploaded with curldrop here:
http://use.yt/upload/6f07d016

to see this error, I ran: 
 (I don't think using a trident field here matters much, but it  could. Ignore the trident YTFieldNotFound error for now, that isn't the problem.)

```
import yt
import trident
ds = yt.load('snap_M1196_4x_conv_470.hdf5')
trident.add_ion_fields(ds, ["H I"])
direction = 'z'
p = yt.ProjectionPlot(ds, direction, ('gas', 'H_p0_number_density'),center = "m",width = (1000, "kpc"))
p.show("no-ticks")
p2 = yt.ProjectionPlot(ds, direction, ('gas', 'H_p0_number_density'),center = "m",width = (500, "kpc"))
p2.show("ticks")
```

I have the dev version of yt installed as described in https://nbviewer.jupyter.org/url/trident-project.org/agora/yt_Trident_AGORA_Tutorial.ipynb#Installing-the-beta-%22demeshened%22-version-of-yt but I've had this problem before when I had the older version, too.

**Comment by ngoldbaum** [comment]
I chatted with @claytonstrawn on slack - it turns out he's using trident 0.5.1-dev. On the latest version of trident the call to `trident.add_ion_fields` crashes for this dataset with the following traceback:

```
--------------------------------------------------------------------------
YTFieldNotFound                           Traceback (most recent call last)
~/Documents/yt-git/yt/mods.py in <module>()
----> 1 trident.add_ion_fields(ds, ["H I"])

~/Documents/trident/trident/ion_balance.py in add_ion_fields(ds, ions, ftype, ionization_table, field_suffix, line_database, force_override, particle_type)
    297         add_ion_mass_field(atom, ion, ds, ftype, ionization_table,
    298             field_suffix=field_suffix, force_override=force_override,
--> 299             particle_type=particle_type)
    300
    301 def add_ion_fraction_field(atom, ion, ds, ftype="gas",

~/Documents/trident/trident/ion_balance.py in add_ion_mass_field(atom, ion, ds, ftype, ionization_table, field_suffix, force_override, particle_type)
    808     ds.add_field((ftype, field), function=_ion_mass, units=r"g",
    809                  particle_type=particle_type,
--> 810                  force_override=force_override)
    811     if ion == 1: # add aliased field too
    812         ds.field_info.alias((ftype, alias_field), (ftype, field))

~/Documents/yt-git/yt/data_objects/static_output.py in add_field(self, name, function, sampling_type, **kwargs)
   1234         self.field_info.add_field(name, sampling_type, function=function, **kwargs)
   1235         self.field_info._show_field_errors.append(name)
-> 1236         deps, _ = self.field_info.check_derived_fields([name])
   1237         self.field_dependencies.update(deps)
   1238

~/Documents/yt-git/yt/fields/field_info_container.py in check_derived_fields(self, fields_to_check)
    383             fi = self[field]
    384             try:
--> 385                 fd = fi.get_dependencies(ds = self.ds)
    386             except Exception as e:
    387                 if field in self._show_field_errors:

~/Documents/yt-git/yt/fields/derived_field.py in get_dependencies(self, *args, **kwargs)
    217             e.requested.append(self.name)
    218         else:
--> 219             e[self.name]
    220         return e
    221

~/Documents/yt-git/yt/fields/field_detector.py in __missing__(self, item)
     97         if finfo is not None and finfo._function.__name__ != 'NullFunc':
     98             try:
---> 99                 vv = finfo(self)
    100             except NeedsGridType as exc:
    101                 ngz = exc.ghost_zones

~/Documents/yt-git/yt/fields/derived_field.py in __call__(self, data)
    243                 "for %s" % (self.name,))
    244         with self.unit_registry(data):
--> 245             dd = self._function(self, data)
    246         for field_name in data.keys():
    247             if field_name not in original_fields:

~/Documents/trident/trident/ion_balance.py in _ion_mass(field, data)
    842         fraction_field_name = "%s_ion_fraction%s" % (prefix, suffix)
    843         return data[ftype, fraction_field_name] * \
--> 844           data[ftype, "particle_mass"]
    845     else:
    846         return data[ftype, density_field_name] * \

~/Documents/yt-git/yt/fields/field_detector.py in __missing__(self, item)
     87         else:
     88             field = item
---> 89         finfo = self.ds._get_field_info(*field)
     90         # For those cases where we are guessing the field type, we will
     91         # need to re-update -- otherwise, our item will always not have the

~/Documents/yt-git/yt/data_objects/static_output.py in _get_field_info(self, ftype, fname)
    775                     self._last_finfo = self.field_info[(ftype, fname)]
    776                     return self._last_finfo
--> 777         raise YTFieldNotFound((ftype, fname), self)
    778
    779     def _setup_classes(self):

YTFieldNotFound: Could not find field '('gas', 'particle_mass')' in snap_M1196_4x_conv_470.
```

This can be fixed on the trident side with the following patch:

```
diff --git a/trident/ion_balance.py b/trident/ion_balance.py
--- a/trident/ion_balance.py
+++ b/trident/ion_balance.py
@@ -840,8 +840,12 @@ def _ion_mass(field, data):
     density_field_name = "%s_density%s" % (prefix, suffix)
     if data.ds.field_info[(ftype, density_field_name)].particle_type:
         fraction_field_name = "%s_ion_fraction%s" % (prefix, suffix)
+        if ftype == 'gas':
+            mass_name = "mass"
+        else:
+            mass_name = "particle_mass"
         return data[ftype, fraction_field_name] * \
-          data[ftype, "particle_mass"]
+            data[ftype, mass_name]
     else:
         return data[ftype, density_field_name] * \
           data[ftype, "cell_volume"]
```

Depending on what @chummels thinks we may actually just want to define both `('gas', 'mass')` and `('gas', 'particle_mass)` for SPH data in yt to avoid the need for trident and other code in the wild to adapt.

Unfortunately after that I get another error that I need to look into closer. Unfortunately both of these are completely unrelated to the weird behavior of the colorbars in Clayton's example, which I'm still not able to reproduce :/

**Comment by ngoldbaum** [comment]
I was able to reproduce this and think this is due to a bug in matplotlib. I've filed an issue over there:

https://github.com/matplotlib/matplotlib/issues/10051

**Comment by ngoldbaum** [comment]
Closing since this will be fixed upstream in matplotlib 3.0

**Comment by ngoldbaum** [comment]
See https://github.com/matplotlib/matplotlib/issues/10051
